pkg/controller/sparkapplication/sparkapp_util.go

```diff
@@ -72,6 +72,22 @@ func getResourceLabels(app *v1beta2.SparkApplication) map[string]string {
 	return labels
 }
 
+func getResourceAnnotations(app *v1beta2.SparkApplication) map[string]string {
+	annotations := map[string]string{}
+
+	if app.Spec.UIConfig != nil {
+		if app.Spec.UIConfig.IngressClass != nil {
+			if *app.Spec.UIConfig.IngressClass == "nginx" {
+				annotations["kubernetes.io/ingressclass"] = "nginx"
+				if app.Spec.UIConfig.EnableSSL && app.Spec.UIConfig.ForceSSL {
```

Comment:
Will enabling and forcing SSL only apply to ingress backed by `nginx`?

Reply:
I'm not sure, i only know that other IngressClasses exist, but i don't know how they behave. But i opened the door with this function, and anyone that well know other cases will be able to append it.